.github/workflows/test.yml

```
@@ -1,32 +1,45 @@
 name: Test
 
 on:
-  - pull_request
+  pull_request: {}
+  push:
+    branches:
+      - master
+
+env:
+  BUNDLE_WITHOUT: release
 
 jobs:
-  test:
+  rspec:
     runs-on: ubuntu-latest
     strategy:
-      fail-fast: false
+      fail-fast: true
       matrix:
-        ruby:
-          - "2.4"
-          - "2.5"
-          - "2.6"
-          - "2.7"
+        include:
+          - ruby: "2.7"
+            coverage: "yes"
     env:
-      BUNDLE_WITHOUT: release
-    name: Ruby ${{ matrix.ruby }}
+      COVERAGE: ${{ matrix.coverage }}
+    name: RSpec - Ruby ${{ matrix.ruby }}
     steps:
-      - uses: actions/checkout@v2
+      - uses: actions/checkout@v3
       - name: Install Ruby ${{ matrix.ruby }}
         uses: ruby/setup-ruby@v1
         with:
           ruby-version: ${{ matrix.ruby }}
           bundler-cache: true
-      - name: Run spec tests
+      - name: spec tests
         run: bundle exec rake test:spec
+      - name: Build gem
+        run: gem build *.gemspec
       # It seems some additonal setup of Docker may be needed for
       # the acceptance tests to work.
       # - name: Run acceptance tests
       #   run: bundle exec rake test:acceptance
+  tests:
```

I'd visually separate this
```suggestion

  tests:
```